src/pydvl/valuation/samplers/base.py

```diff
@@ -113,14 +112,33 @@ def from_data(self, data: Dataset) -> BatchGenerator:
 
     def from_indices(self, indices: IndexSetT) -> BatchGenerator:
         """Batches the samples and yields them."""
+
+        # early return for empty indices. Necessary because some samplers use a
+        # while True loop to generate infinite samples.
+        if len(indices) == 0:
+            return
+
         self._interrupted = False
+        self._n_samples = 0
         for batch in take_n(self._generate(indices), self.batch_size):
             yield batch
             # FIXME, BUG: this could be wrong if the batch is not full. Just use lists
             self._n_samples += self.batch_size
             if self._interrupted:
                 break
 
+    def length(self, indices: IndexSetT) -> int | None:
```

Comment:
Given what is happening here, maybe sample_limit could be a better name?

Reply:
I agree, done!